code/modules/power/engines/fission/nuclear_misc.dm

```diff
@@ -0,0 +1,896 @@
+
+			if(!materials.use_amount(materials_to_use))
+				to_chat(usr, SPAN_WARNING("Failed to deduct materials!"))
+				return FALSE
+
+			// Begin Process
+			begin_fabrication(rod_type_path)
+
+			return TRUE
+
+		if("eject_material")
+			var/material_id = params["id"]
+			var/amount = params["amount"]
+
+			if(!material_id || !materials.materials[material_id])
+				return FALSE
+
+			var/desired_sheets = 0
+			if(amount == "custom")
+				var/datum/material/M = materials.materials[material_id]
+				var/max_sheets = round(M.amount / MINERAL_MATERIAL_AMOUNT)
+				if(max_sheets <= 0)
+					to_chat(usr, SPAN_WARNING("Not enough [M.name] to eject!"))
+					return FALSE
+				desired_sheets = tgui_input_number(usr, "How many sheets do you want to eject?", "Ejecting [M.name]", 1, max_sheets, 1)
+				if(isnull(desired_sheets))
+					return FALSE
+			else
+				desired_sheets = text2num(amount)
+
+			desired_sheets = max(0, round(desired_sheets))
+			if(desired_sheets > 0)
+				materials.retrieve_sheets(desired_sheets, material_id, get_turf(src))
+				to_chat(usr, SPAN_NOTICE("[src] ejects [desired_sheets] sheets."))
+
+			return TRUE
+
+	return FALSE
+
+/obj/machinery/nuclear_rod_fabricator/screwdriver_act(mob/user, obj/item/I)
+	. = TRUE
+	if(power_state == ACTIVE_POWER_USE)
+		to_chat(user, SPAN_WARNING("The machine cannot be opened while it is operating."))
+		return ITEM_INTERACT_COMPLETE
+	default_deconstruction_screwdriver(user, "rod_fab_maint", "rod_fab", I)
+
+
+/obj/machinery/nuclear_rod_fabricator/proc/begin_fabrication(rod_type_path)
+	power_state = ACTIVE_POWER_USE
+	icon_state = "rod_fab_on"
+	COOLDOWN_START(src, fabrication_timer, work_time)
+	schematic = rod_type_path
+
+/obj/machinery/nuclear_rod_fabricator/proc/finish_fabrication()
+	power_state = IDLE_POWER_USE
+	icon_state = "rod_fab"
+	var/obj/item/nuclear_rod/new_rod = new schematic(get_turf(src))
+	to_chat(usr, SPAN_NOTICE("[src] fabricates \a [new_rod.name]."))
+	playsound(src, 'sound/machines/ping.ogg', 50, 1)
+
+/obj/machinery/nuclear_rod_fabricator/proc/abort_fabrication()
+	power_state = IDLE_POWER_USE
+	icon_state = "rod_fab"
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+
+/obj/machinery/nuclear_rod_fabricator/process()
+	if(stat & NOPOWER)
+		if(power_state == ACTIVE_POWER_USE)
+			abort_fabrication()
+		return
+	if(power_state == IDLE_POWER_USE)
+		return
+	if(!COOLDOWN_FINISHED(src, fabrication_timer))
+		return
+	finish_fabrication()
+
+// MARK: Reactor Power Output
+
+/obj/machinery/power/reactor_power
+	name = "reactor output terminal"
+	desc = "A bundle of heavy watt power cables for managing power output from the reactor."
+	icon_state = "term"
+	plane = FLOOR_PLANE
+	layer = WIRE_TERMINAL_LAYER //a bit above wires
+	resistance_flags = INDESTRUCTIBLE
+	var/power_gen = 0
+	var/obj/machinery/atmospherics/fission_reactor/linked_reactor
+
+/obj/machinery/power/reactor_power/Initialize(mapload)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
```

Comment:
it's fine to push some setup to late load but could you comment why it's necessary since it's the less usual case?